### ISSUE TITLE
Links/Nodes files being written into EPSG:4326

### ISSUE THREAD

**Issue by Lvulis** [jonschwenk/RivGraph#6]
In some sample data I have a delta class with EPSG: 32603 (UTM Zone 3N), but the nodes and link that are written are in EPSG 4326 (unprojected). The skeleton is written to 32603. Not sure if this matters in pruning the nodes/links, but it makes visual inspection/overlaying of the skeleton, nodes, and links data in QGIS a multistep process. Found during debugging/probing why prune network produces an error while removing spurs, will follow up in another issue if necessary.

```python
yukon.epsg
Out[40]: 32603

yukon.to_geovectors('network', ftype='json') # ftype can be either 'shp' or 'json'

C:\Anaconda3\envs\geo_py37\lib\site-packages\pyproj\crs\crs.py:53: FutureWarning: '+init=<authority>:<code>' syntax is deprecated. '<authority>:<code>' is the preferred initialization method. When making the change, be mindful of axis order changes: https://pyproj4.github.io/pyproj/stable/gotchas.html#axis-order-changes-in-proj-6
  return _prepare_from_string(" ".join(pjargs))

```

Data:
[Yukon_data.zip](https://github.com/jonschwenk/RivGraph/files/4389750/Yukon_data.zip)

**Comment by jonschwenk** [comment]
This is another manifestation of crs() handling in the newer version of pyproj/fiona. If you want to avoid these errors, downgrade pyproj to 2.2.1. I cannot reproduce this issue with 2.2.1. It might wreck your anaconda environment to do it, so feel free to patch it however you can for now. I'll move either (a) providing "bulletproof" install instructions or (b) updating RivGraph to mesh with newer pyproj versions higher up the priority list. I guess they're not mutually exclusive...

**Comment by Lvulis** [comment]
OK - for reference:
```python
pyproj.__version__
Out[57]: '2.6.0'

geopandas.__version__
Out[59]: '0.7.0'

fiona.__version__
Out[60]: '1.8.13'
```

**Comment by jonschwenk** [comment]
When you instantiate your `delta `or `river` class, rivgraph reads the EPSG code from the input mask and stores that. You can check it with `delta_class_object.epsg`. Yours will say 32603. This EPSG code is used for all rivgraph exports, so when functioning correctly all exports (rasters or vectors) will have this CRS. 

Don't try to change the epsg manually. Probably the best workaround for you is to add a couple lines that load in the exported geojson (or shapefile), assign it the correct projection, then write it back out. Should be two lines with geopandas, but apparently the syntax is now different.

**Comment by Lvulis** [comment]
Got it. Thanks Jon.

**Comment by jonschwenk** [comment]
I'm going to leave this issue Open as a "catch all" for the other related issues you opened. The solution is to rewrite rivgraph to remove the deprecated syntaxes of pyproj.

#4 
#3

**Comment by Lvulis** [comment]
General question but how frequent are these update cycles for the underlying packages and why is there so much instability and incompatibility between package versions? I see for example your version of geopandas is only from last July!

**Comment by jonschwenk** [comment]
I'm not really sure. I've only been python-ing for 3 years now. I think frequency of updates is very package- and developer-specific. More developers usually means more updates, and it's a lot of work to keep your packages (rivgraph in this case) compatible with the latest versions of everything. Hence the need for good unit tests! And a team of developers...

For a package like rivgraph, I think it's acceptable to keep it "locked in" to outdated packages **provided** bulletproof environment setup instructions. With virtual environments, it's easy enough to create a rivgraph-specific environment, activate it, and do your analysis. If you want to use incompatible packages to analyze rivgraph outputs, you can just create another environment. Not ideal, but considering the heavy lifting that rivgraph does, probably worth it. Anyway, I'll try to fix this particular issue.

**Comment by Lvulis** [comment]
That solution would work well for me and I have a feeling for Alex as well.

**Comment by Lvulis** [comment]
This is another related bug that comes up because of the changes to subsetting. My workaround is in the 2nd codeblock. 

```python
  File "C:\Anaconda3\envs\geo_py37\lib\site-packages\rivgraph-0.2-py3.7.egg\rivgraph\deltas\delta_utils.py", line 150, in clip_by_shoreline
    if links_gdf.crs['init'] != shore_gdf.crs['init']:

TypeError: 'CRS' object is not subscriptable
```

```python
 links_gdf_epsg = links_gdf.crs.to_epsg()
    shore_gdf_epsg = shore_gdf.crs.to_epsg()
    # Enusre we have consistent CRS before intersecting
    # if links_gdf.crs['init'] != shore_gdf.crs['init']:
    if links_gdf_epsg - shore_gdf_epsg != 0:
        shore_gdf = shore_gdf.to_crs(links_gdf.crs)
    shore_gdf.crs = links_gdf.crs #  This line is to prevent the "crs do not match" warning--the warning is triggered by "nodefs: True" in some crs dicts, not the actual CRS as it ensured to be the same in the previous two lines
               
```

**Comment by jonschwenk** [comment]
Just leaving this here as a reference for the new syntax. Geopandas 0.7 is when the new syntax was implemented: https://geopandas.org/projections.html, which agrees with our posted versions above.

**Comment by Lvulis** [comment]
Awesome. Looking forward to any RG updates. :)

**Comment by jonschwenk** [comment]
This issue has been fixed as of version 0.3, which pins RivGraph to geopandas 0.7. @Lvulis if you check out the [main page](https://github.com/jonschwenk/RivGraph), you'll find updated install instructions that are quick and easy and include some minor bugfixes.
